ddtrace/ext/aws.py

```diff
@@ -27,6 +28,33 @@ def unpacking_args(args, args_name, traced_args_list):
     return response
 
 
+def flatten_args(args):
+    """Filter operation arguments based on list of arguments to be traced
+    """
+
+    # unnest structure for argument values that are not simple/atomic
+    def _unnest(t):
+        key, value = t
+        if isinstance(value, dict):
+            return [
+                ('{}.{}'.format(key, inner_key), inner_value)
+                for (inner_key, inner_value) in value.items()
+            ]
+        else:
+            return (key, value)
```

Comment:
If you have this do `return [(key, value)]` then your logic below is actually easier.


```python
return dict(
    (key, value)
    for t in args
    for (key, value) in _unnest(t)
)
```

Reply:
Added a `flatten_dict` function to simplify the setting tags from the operation arguments.